tests/basic_one_way_send.go

```diff
@@ -0,0 +1,175 @@
+	"math/big"
+	"time"
+
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	warpBackend "github.com/ava-labs/subnet-evm/warp"
+	. "github.com/ava-labs/teleporter/tests/utils"
+
+	"github.com/ava-labs/awm-relayer/messages/teleporter"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/interfaces"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ava-labs/subnet-evm/x/warp"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	teleporterMessage = teleporter.TeleporterMessage{
+		MessageID:               big.NewInt(1),
+		SenderAddress:           FundedAddress,
+		DestinationAddress:      FundedAddress,
+		RequiredGasLimit:        big.NewInt(1),
+		AllowedRelayerAddresses: []common.Address{},
+		Receipts:                []teleporter.TeleporterMessageReceipt{},
+		Message:                 []byte{1, 2, 3, 4},
+	}
+)
+
+// Ginkgo describe node that acts as a container for the teleporter e2e tests. This test suite
+// will run through the following steps in order:
+// 1. Send a transaction to the Teleporter contract on Subnet A
+// 2. Aggregate signatures and send the Warp message to Subnet B
+// 3. Verify receipt of the message on Subnet B
+var _ = ginkgo.Describe("[Teleporter one way send]", ginkgo.Ordered, func() {
+	var (
+		teleporterMessageID *big.Int
+	)
+
+	// Send a transaction to Subnet A to issue a Warp Message from the Teleporter contract to Subnet B
+	ginkgo.It("Send Message from A to B", ginkgo.Label("Teleporter", "SendTeleporter"), func() {
+		ctx := context.Background()
+
+		nonceA, err := ChainARPCClient.NonceAt(ctx, FundedAddress, nil)
+		Expect(err).Should(BeNil())
+
+		data, err := teleporter.EVMTeleporterContractABI.Pack(
+			"sendCrossChainMessage",
+			SendCrossChainMessageInput{
+				DestinationChainID: BlockchainIDB,
+				DestinationAddress: FundedAddress,
+				FeeInfo: FeeInfo{
+					ContractAddress: FundedAddress,
+					Amount:          big.NewInt(0),
+				},
+				RequiredGasLimit:        big.NewInt(1),
+				AllowedRelayerAddresses: []common.Address{},
+				Message:                 []byte{1, 2, 3, 4},
+			},
+		)
+		Expect(err).Should(BeNil())
+
+		// Send a transaction to the Teleporter contract
+		tx := NewTestTeleporterTransaction(ChainAIDInt, TeleporterContractAddress, nonceA, data)
+
+		txSigner := types.LatestSignerForChainID(ChainAIDInt)
+		signedTx, err := types.SignTx(tx, txSigner, FundedKey)
+		Expect(err).Should(BeNil())
+
+		subA, err := ChainAWSClient.SubscribeNewHead(ctx, NewHeadsA)
+		Expect(err).Should(BeNil())
+		defer subA.Unsubscribe()
+
+		log.Info("Sending Teleporter transaction on source chain", "destinationChainID", BlockchainIDB, "txHash", signedTx.Hash())
+		err = ChainARPCClient.SendTransaction(ctx, signedTx)
+		Expect(err).Should(BeNil())
+
+		// Sleep to ensure the new block is published to the subscriber
+		time.Sleep(5 * time.Second)
```

Comment:
I think we could speed the test suite up quite a bit if we created a helper function `WaitNewBlock` that takes an RPC client as a param, and calls `SubscribeNewHead` and waits for one new block to be produced. That way we can eliminate the `time.Sleep`s.

Reply:
I agree we should be waiting on new blocks rather than sleeps. However, a helper might be difficult to set up properly since we have to subscribe before sending a transaction in order to ensure that the new block will be picked up by the subscription.

Reply:
`SendAndWaitForTransaction` is implemented as a wrapper around `ethclient.SendTransaction`. One case that didn't quite fit is the call to deploy the Teleporter contract raw tx bytes, but that's a one off case so I think it's fine that this helper doesn't apply.